### ISSUE TITLE
Unable to hide comments if saving using data that already contains VML drawings.

### ISSUE THREAD

**Issue by benstilwell** [PHPOffice/PhpSpreadsheet#4105]
This is:

```
- [*] a bug report
```

### What is the expected behavior?
When saving a modified spreadsheet with legacyDrawing as xlsx. I expect to still be able to save newly created VMLComments.

### What is the current behavior?
Currently when saving a check is made for an existing legacyDrawing and as a result writeVMLComments is never called. This makes it impossible to hide newly created comments.

### What are the steps to reproduce?

Have an existing xlsx file containing VML drawings.
Load file and add comments
set visibility of comments to false
Save file and when loading in an excel file reader, the comments are not hidden

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
 $excelReader = \PhpOffice\PhpSpreadsheet\IOFactory::createReader('Xlsx');
 $excelData = $excelReader->load($fileWithLegacyDrawing);

$worksheet = $excelData->getSheet(0);
$worksheet->getComment('A2')->getText()->createText("Just Hide Me");      
$worksheet->getComment('A2')->setVisible(false);

$excelWriter = \PhpOffice\PhpSpreadsheet\IOFactory::createWriter($excelData, 'Xlsx');
   
$excelWriter->save($newFile);

```

### What features do you think are causing the issue

- [ ] Reader
- [*] Writer
- [ ] Styles
- [ ] Data Validations
- [ ] Formula Calculations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?
Only tested with Xlsx

### Which versions of PhpSpreadsheet and PHP are affected?
Currently using php 8,2, PhpSpreadsheet 2.1.0

**Comment by oleibman** [comment]
The handling of legacyDrawing is a consequence of PR #3130, which enabled the passing through of form controls if a spreadsheet was loaded and subsequently saved. Your problem is an unintended consequence of that change for spreadsheets with comments but without form controls. There are at least 2 possible approaches to rectifying it. Adding the ability to remove legacyDrawings from worksheets is probably pretty easy. Analyzing the legacyDrawings and automatically removing them if they contain only notes (with no form control elements) is a lot more difficult, but probably a more sensible way to approach the problem. I will add it to my to-do list, but I may not be able to get to it quickly.

**Comment by oleibman** [comment]
Should be fixed by PR #4122, which is now merged.
